dashboard/src/actions/repos.ts

```diff
@@ -175,6 +183,25 @@ export const installRepo = (
   };
 };
 
+export const validateRepo = (
+  repoURL: string,
+  authHeader: string,
+  customCA: string,
+): ThunkAction<Promise<boolean>, IStoreState, null, AppReposAction> => {
+  return async dispatch => {
+    try {
+      dispatch(checkRepo());
+      const data = await AppRepository.validate(repoURL, authHeader, customCA);
+      dispatch(validatedRepo(data));
+
+      return true;
+    } catch (e) {
```

Comment:
The thing is that `AppRepository.validate` throws an error if the validation fails. I can add a `return data === "OK"` for the sake of readability but if the call doesn't fail, it's because the validation succeeded.